sql/src/main/java/io/crate/expression/operator/any/AnyLikeOperator.java

```diff
@@ -84,36 +78,46 @@ private Boolean doEvaluate(Object left, Iterable<?> rightIterable) {
             }
             assert elem instanceof String : "elem must be a String";
             String elemValue = (String) elem;
-            if (matches.test(elemValue, pattern)) {
+            if (matches.test(elemValue, pattern, ignoreCase)) {
                 return true;
             }
         }
         return hasNull ? null : false;
     }
 
     @Override
-    public Boolean evaluate(TransactionContext txnCtx, Input<Object>... args) {
-        Object value = args[0].value();
+    public Boolean evaluate(TransactionContext txnCtx, Input<String>... args) {
+        String value = args[0].value();
         Object collectionReference = args[1].value();
-
+        boolean ignoreCase = (Boolean) ((Object) args[2].value());
         if (collectionReference == null || value == null) {
             return null;
         }
-        return doEvaluate(value, collectionValueToIterable(collectionReference));
+        return doEvaluate(value, collectionValueToIterable(collectionReference), ignoreCase);
     }
 
-    private static class AnyLikeResolver extends BaseFunctionResolver {
+    @FunctionalInterface
+    private interface TriPredicate<A, B, C> {
+        boolean test(A a, B b, C c);
+
+        static <A, B, C> TriPredicate<A, B, C> negate(TriPredicate<A, B, C> pred) {
+            return pred.negate();
+        }
+
+        default TriPredicate<A, B, C> negate() {
+            return (A a, B b, C c) -> false == test(a, b, c);
```

Comment:
Maybe inline this into the static method above as it is the only place we use it.